Handle deleted worktree directories gracefully without error logging

## Summary

When a worktree directory is deleted externally (e.g., after merging a PR or manually removing a worktree), Canopy logs verbose `[ERROR]` messages to stderr even though this is an expected, recoverable scenario. These errors should be handled silently or logged at warn/debug level instead.

## Current Behavior

When monitoring a worktree that gets deleted:

```
[ERROR] Git worktree changes operation failed {
  "message": "Failed to get git worktree changes",
  "name": "GitError",
  "stack": "GitError: Failed to get git worktree changes\n    at getWorktreeChangesWithStats ...",
  "context": { "cwd": "/path/to/deleted-worktree" },
  "cause": {
    "message": "fatal: Unable to read current working directory: No such file or directory\n"
  }
}
[ERROR] Failed to update git status { ... }
```

This happens because:
1. `getWorktreeChangesWithStats()` correctly detects the missing directory and throws `WorktreeRemovedError`
2. But the outer try/catch wraps it in a `GitError` and logs `[ERROR]` before the caller can handle it
3. The error reaches `WorktreeMonitor.updateGitStatus()` which handles `WorktreeRemovedError` gracefully
4. But by then, the error has already been logged

## Expected Behavior

- No `[ERROR]` logs for deleted worktree directories (this is a normal lifecycle event)
- The `WorktreeRemovedError` should propagate without being wrapped/logged at the git.ts level
- `WorktreeMonitor` should handle it silently (it already does, but too late)
- Consider logging at `debug` level for diagnostic purposes only

## Steps to Reproduce

1. Start Canopy in a project with multiple worktrees
2. In another terminal, delete one of the worktrees: `git worktree remove branch-name`
3. Observe `[ERROR]` messages in Canopy's stderr output
4. The worktree card disappears correctly, but the error logs are confusing

**Reproducibility**: Always

## Environment

- **OS**: macOS, Linux (any)
- **Node.js**: 20.19.0+
- **Canopy**: Current main branch

## Evidence

Error occurs at:
- [`src/utils/git.ts:403`](https://github.com/gregpriday/canopy/blob/main/src/utils/git.ts#L403) - logs `[ERROR] Git worktree changes operation failed`
- Caught at [`src/services/monitor/WorktreeMonitor.ts:573`](https://github.com/gregpriday/canopy/blob/main/src/services/monitor/WorktreeMonitor.ts#L573) - handles `WorktreeRemovedError` correctly

**Root Cause**: The `WorktreeRemovedError` thrown at git.ts:163 is caught by the outer catch block at git.ts:396 which re-wraps it as a generic `GitError` and logs an error before re-throwing.

## Deliverables

### Code Changes

**Files to Modify:**
- [`src/utils/git.ts`](https://github.com/gregpriday/canopy/blob/main/src/utils/git.ts) - Re-throw `WorktreeRemovedError` without wrapping or logging

### Tests

- Add test case verifying `WorktreeRemovedError` propagates cleanly without being wrapped
- Verify no `logError` calls for deleted worktree scenarios

## Technical Specifications

**Footprint:** `src/utils/git.ts`, `src/services/monitor/WorktreeMonitor.ts`

**Approach:** In `getWorktreeChangesWithStats()`, check if the caught error is already a `WorktreeRemovedError` and re-throw it directly without wrapping in `GitError` or logging.

## Dependencies

None

## Tasks

- [ ] Modify catch block in [`src/utils/git.ts:396-405`](https://github.com/gregpriday/canopy/blob/main/src/utils/git.ts#L396-L405) to re-throw `WorktreeRemovedError` without wrapping
- [ ] Remove or downgrade the `logError` call for `WorktreeRemovedError` scenarios
- [ ] Add unit test for error propagation behavior
- [ ] Verify fix with manual testing (delete worktree while Canopy is running)

## Acceptance Criteria

- [ ] Deleting a worktree externally produces no `[ERROR]` output in stderr
- [ ] `WorktreeRemovedError` reaches `WorktreeMonitor` without being wrapped in `GitError`
- [ ] Worktree card still disappears correctly after deletion
- [ ] Debug/warn logging still available via `CANOPY_DEBUG=1` if needed
- [ ] Existing tests pass

## Edge Cases & Risks

- **Transient ENOENT**: File might be temporarily inaccessible (heavy I/O, network filesystem). Current behavior keeps monitor running for recovery, which is correct.
- **Permission errors**: Should still log errors for EACCES/EPERM since those indicate real problems.
- **Other git errors**: Must continue logging actual git failures (e.g., corrupt repo, index.lock conflicts).

## Additional Context

The user reported this during normal development workflow - worktrees are frequently created/deleted when working on issues. The error spam makes it hard to see actual problems.

--- a/src/utils/git.ts
+++ b/src/utils/git.ts
@@ -394,6 +394,23 @@ export async function getWorktreeChangesWithStats(
     GIT_WORKTREE_CHANGES_CACHE.set(cwd, result);
     return result;
   } catch (error) {
+    // Re-throw WorktreeRemovedError without wrapping or logging
+    // This is an expected lifecycle event (worktree deleted externally), not an error
+    if (error instanceof WorktreeRemovedError) {
+      throw error;
+    }
+
+    // Handle race condition: directory disappeared between fs.access check and git operations
+    // simple-git errors contain the ENOENT message, convert to WorktreeRemovedError
+    const errorMessage = error instanceof Error ? error.message : String(error);
+    if (
+      errorMessage.includes('ENOENT') ||
+      errorMessage.includes('no such file or directory') ||
+      errorMessage.includes('Unable to read current working directory')
+    ) {
+      throw new WorktreeRemovedError(cwd, error instanceof Error ? error : undefined);
+    }
+
     const cause = error instanceof Error ? error : new Error(String(error));
     const gitError = new GitError(
       'Failed to get git worktree changes',

--- a/tests/utils/git.test.ts
+++ b/tests/utils/git.test.ts
@@ -345,6 +345,47 @@ describe('git.ts', () => {
 			expect(result.changes).toHaveLength(1);
 			expect(result.changes[0].status).toBe('modified');
 		});
+
+		it('propagates WorktreeRemovedError without wrapping in GitError', async () => {
+			// Mock fs.access to fail with ENOENT
+			const enoentError = new Error('ENOENT: no such file or directory');
+			(enoentError as NodeJS.ErrnoException).code = 'ENOENT';
+			vi.mocked(mockAccess).mockRejectedValue(enoentError);
+
+			try {
+				await getWorktreeChangesWithStats('/deleted/worktree', true);
+				expect.fail('Should have thrown');
+			} catch (error) {
+				// Error should be exactly WorktreeRemovedError, not wrapped in GitError
+				expect(error).toBeInstanceOf(WorktreeRemovedError);
+				// The error message should be the WorktreeRemovedError message, not GitError's
+				expect((error as Error).message).toBe('Worktree directory no longer exists');
+				// Verify it's not wrapped (no "Failed to get git worktree changes" message)
+				expect((error as Error).message).not.toContain('Failed to get git worktree changes');
+			}
+		});
+
+		it('converts git ENOENT errors to WorktreeRemovedError (race condition)', async () => {
+			// Mock fs.access to succeed (directory exists at check time)
+			vi.mocked(mockAccess).mockResolvedValue(undefined);
+
+			// But then simple-git fails because directory disappeared mid-operation
+			const simpleGit = await import('simple-git');
+			vi.mocked(simpleGit.default).mockReturnValue({
+				status: vi.fn().mockRejectedValue(
+					new Error('fatal: Unable to read current working directory: No such file or directory')
+				),
+			} as any);
+
+			try {
+				await getWorktreeChangesWithStats('/disappearing/worktree', true);
+				expect.fail('Should have thrown');
+			} catch (error) {
+				// Should be converted to WorktreeRemovedError, not wrapped in GitError
+				expect(error).toBeInstanceOf(WorktreeRemovedError);
+				expect((error as Error).message).toBe('Worktree directory no longer exists');
+			}
+		});
 	});
 
 	describe('getCommitCount', () => {

